app/expences.py

```diff
@@ -0,0 +1,84 @@
+import math
+import datetime
+
+from app.customers import Customer
+from app.shops import Shop
+
+
+def find_min_cost(
+        customer: Customer, shops: list[Shop], fuel_price: float
+) -> tuple | None:
+    budget = customer.money
+    print(f"{customer.name} has {budget} dollars")
+    (shop_head_to,
+     total_cost,
+     purchase_cost_min,
+     _distance
+     ) = [shops[0], 0, 0, 0]
+    for shop in shops:
+        purchase_cost = sum(
+            [shop.products[product] * customer.product_cart[product]
+             for product in customer.product_cart]
+        )
```

Comment:
You could pass a generator expression instead of list comprehension to the `sum`.